### ISSUE TITLE
Implement 'experimental' flag

### ISSUE THREAD

**Issue by dylanb** [dequelabs/axe-core#140]
As new specs are developed, we should have a way of running rules that reflect these emerging standards.

Example: aria-checkbox is allowed to have aria-readonly in ARIA 1.1 but not in ARIA 1.0, this conflicting requirement should not be on by default.
 
Implementing this feature will require some updates to the tag system to allow filters to be applied (currently tags are union only, we require combinations of union and intersection for this to work properly)

**Comment by WilcoFiers** [comment]
I've looked at this issue and I'd suggest using the following syntax for this:

```javascript
// Default: wcag2a + wcag2aa, without experimental
runOnly: {
	type: 'tags',
	value: {
		include: ['wcag2a', 'wcag2aa'],
		exclude: ['experimental']
	}
}

// Get all level A including experimental
runOnly: {
	type: 'tags',
	value: {
		include: ['wcag2a', 'experimental'],
		exclude: ['wcag2aa', 'wcag2aaa']
	}
}

// Get only level AA experimental:
runOnly: {
	type: 'tags',
	value: {
		include: [experimental],
		exclude: ['wcag2a', 'wcag2aaa']
	}
}
```

First all includes are added together. Everything matching exclude is then filtered out.

**Comment by dylanb** [comment]
Is the logic the following?

1. Find all the rules that match the `include` list, then
2. Filter out the rules that match the `exclude` list, then
3. Execute the resulting list of rules

**Comment by WilcoFiers** [comment]
Yes, that's how I want to do it.

**Comment by dylanb** [comment]
seems ok to me :-)

**Comment by WilcoFiers** [comment]
Resolved in the internal repo. Will be available in the next major release of axe-core.
